components/core/src/clp/ffi/ir_stream/Deserializer.hpp

```diff
@@ -20,37 +20,62 @@
 #include "IrUnitHandlerInterface.hpp"
 #include "IrUnitType.hpp"
 #include "protocol_constants.hpp"
+#include "search/QueryHandlerReq.hpp"
 #include "utils.hpp"
 
 namespace clp::ffi::ir_stream {
 /**
  * A deserializer for reading IR units from a CLP kv-pair IR stream. An IR unit handler should be
- * provided to perform user-defined operations on each deserialized IR unit.
+ * provided to perform user-defined operations on each deserialized IR unit. Additionally, a query
+ * handler can be provided to handle KQL queries and column projections.
```

Comment:
```suggestion
 * handler can be provided to handle queries and column projections.
```

Just in case we rename the query language in the future, it's probably easier to not mention it here.